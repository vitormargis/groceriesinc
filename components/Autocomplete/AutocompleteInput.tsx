import React, { FC, KeyboardEvent, ChangeEvent } from 'react';

interface AutocompleteInputProps {
  input: string,
  preInput: string,
  options: string[],
  onKeyDown: (e: KeyboardEvent<HTMLInputElement>) => void, 
  onChange: (e: ChangeEvent<HTMLInputElement>, options: string[]) => void
}

const AutocompleteInput: FC<AutocompleteInputProps> = ({onChange, onKeyDown, input, preInput, options})  => {
  return <>
    <input
      type="text"
      onChange={(e) => onChange(e, options)}
      onKeyDown={onKeyDown}
      value={input}
      placeholder="Search your fruit"
      className="autocomplete-input"
      data-cy="autocomplete-input"
    />
    <input
      type="text"
      value={preInput}
      className="autocomplete-pre-input"
      data-cy="autocomplete-pre-input"
      readOnly
    />
  </>
}

export default AutocompleteInput; 