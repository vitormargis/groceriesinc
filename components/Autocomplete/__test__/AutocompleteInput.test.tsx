import React from 'react';
import { act, create } from 'react-test-renderer';
import AutocompleteInput from '../AutocompleteInput';

const props = {
  onChange: () => null,
  preInput: '0',
  input: '',
  options: [ 'apple', 'avocado', 'banana' ],
  onKeyDown: () => null
}

const renderComponent = (props: any) => {
  let result: any;
  act(() => {
    result = create(<AutocompleteInput {...props} />);
  });

  return result;
};

describe('AutocompleteInput', () => {
  it('should render with AutocompleteInput', () => {
    const component = renderComponent(props);
    component.root.findByProps({ className: 'autocomplete-input' }).props.onChange()
    expect(component.toJSON).toMatchSnapshot();
  });
});