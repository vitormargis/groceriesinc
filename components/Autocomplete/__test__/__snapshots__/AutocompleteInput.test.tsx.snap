// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteInput should render with AutocompleteInput 1`] = `[Function]`;
